fl4health/clients/basic_client.py

```diff
@@ -64,15 +68,27 @@ def __init__(
             progress_bar (bool): Whether or not to display a progress bar
                 during client training and validation. Uses tqdm. Defaults to
                 False
+            intermediate_checkpoint_dir (Optional[Path]): An optional path to store per round
+                checkpoints.
+            client_name (str): An optional client name that uniquely identifies a client.
+                If not passed, a hash is randomly generated.
         """
 
         self.data_path = data_path
         self.device = device
         self.metrics = metrics
         self.checkpointer = checkpointer
         self.progress_bar = progress_bar
+        self.client_name = client_name if client_name is not None else generate_hash()
 
-        self.client_name = generate_hash()
+        self.per_round_checkpointer: Union[None, PerRoundCheckpointer]
```

Comment:
Since the project requirements have been upgraded to python 3.10, we can also use `|` in place of Union. I would propose `PerRoundCheckpointer | None`, it's less characters, imo equally clear, and removes the need to import Union or Optional which if we're lucky removes a line in the imports.